### PR TITLE
fix: replace tbl_flatten to flatten():totable()

### PR DESCRIPTION
**Pull request by pablobfonseca**
- support nvim < 0.10

### CODE REVIEW

**Review by akinsho — COMMENTED** [review]
Thanks so much for the PR, I've made a comment to hopefully reduce the potential cost of the logic but otherwise looks good to me

File: lua/bufferline/utils/init.lua [hunk 1]
```diff
@@ -275,4 +275,8 @@ else
   M.is_list = vim.tbl_isarray or vim.tbl_islist
 end
 
+function M.tbl_flatten(t)
+  return vim.fn.has("nvim-0.10") == 1 and vim.iter(t):flatten(math.huge):totable() or vim.tbl_flatten(t)
```
[the remark this answers is not in the document]
Comment (pablobfonseca) [reply lua/bufferline/utils/init.lua]: @akinsho done! let me know if that makes sense

[the remark this answers is not in the document]
Comment (pablobfonseca) [reply lua/bufferline/utils/init.lua]: @akinsho I changed to check for the version 0.11 for `flatten` like it's being done on the telescope plugin:
https://github.com/nvim-telescope/telescope.nvim/blob/4aed63995a69e343b068c7469491a8d1592c339f/plugin/telescope.lua#L121

**Review by akinsho — APPROVED** [review]
Thanks LGTM

**Pull request merged**